src/cmp_aus.js

```diff
@@ -95,31 +95,90 @@ const reloadPage = async (page) => {
 };
 
 const checkPrebid = async (page) => {
-	log(`Reloading page: Start`);
+	// --------------- RELOAD PAGE START ---------------------------
+	log(`[TEST 4: RELOAD PAGE] Step start`);
 	const reloadResponse = await page.reload({
 		waitUntil: 'domcontentloaded',
 		timeout: 30000,
 	});
 	if (!reloadResponse) {
-		logError(`Reloading page : Failed`);
+		logError(`[TEST 4: RELOAD PAGE] Reloading page : Failed`);
 		throw 'Failed to refresh page!';
 	}
-	log(`Reloading page: Complete`);
+	log(`[TEST 4: RELOAD PAGE] Step complete`);
+	// --------------- RELOAD PAGE END ---------------------------
+
+	// --------------- BUNDLE START ---------------------------
+	log(`[TEST 4: PREBID BUNDLE] Checking: graun.Prebid.js.commercial.js`);
+	await page.waitForRequest((req) =>
+		req.url().includes('graun.Prebid.js.commercial.js'),
+	);
+	log(`[TEST 4: PREBID BUNDLE] Step start`);
+	// --------------- BUNDLE END ---------------------------
+
+	// --------------- CANADA START ---------------------------
+	log(`[TEST 4: CANADA] Step start`);
+	const currentLocation = await getCurrentLocation(page);
+	if (currentLocation === 'CA') {
+		log('[TEST 4: CANADA] In Canada we do not run Prebid');
+		return Promise.resolve();
+	}
+	log(`[TEST 4: CANADA] Step complete`);
+	// --------------- CANADA END ---------------------------
 
+	// --------------- PAGESKIN START ---------------------------
+	log(`[TEST 4: PAGESKIN] Step start`);
 	const hasPageskin = await page.evaluate(() =>
 		document.body.classList.contains('has-page-skin'),
 	);
 
 	if (hasPageskin) {
-		log('Pageskin detected. Prebid will not run');
+		log('[TEST 4: PAGESKIN] Pageskin detected. Prebid will not run');
 		return Promise.resolve();
 	}
+	log(`[TEST 4: PAGESKIN] Step complete`);
+	// --------------- PAGESKIN END ---------------------------
 
+	// --------------- PUBMATIC START ---------------------------
+	log(`[TEST 4: PUBMATIC] Step start`);
 	const prebidURL =
 		'https://hbopenbid.pubmatic.com/translator?source=prebid-client';
 
 	await page.waitForRequest((req) => req.url().includes(prebidURL));
-	log(`Prebid check: Complete`);
+	log(`[TEST 4: PUBMATIC] Step complete`);
+	// --------------- PUBMATIC END ---------------------------
+
+	// --------------- PBJS START ---------------------------
+	log(`[TEST 4: PBJS] Step start`);
+	const hasPrebid = await page.evaluate(() => window.pbjs !== undefined);
+	if (!hasPrebid) {
+		logError('[TEST 4: PBJS] Prebid.js is not loaded');
+		throw new Error('Prebid.js is missing');
+	}
+	log(`[TEST 4: PBJS] Step complete`);
+	// --------------- PBJS END ---------------------------
+
+	// --------------- BID RESPONSE START ---------------------------
+	log(`[TEST 4: BID RESPONSE] Step start`);
+	const bidResponses = await page.evaluate(() => {
+		if (window.pbjs) {
+			return pbjs.getBidResponses()['dfp-ad--top-above-nav'];
+		} else {
+			return null;
+		}
+	});
+
+	if (bidResponses) {
```

Comment:
bidResponses could still be empty here

Reply:
This one a tricky one, could we log that the bidResponse is empty?

Reply:
This approach seems to work fine for the `hasPrebid` but not so much for this `bidResponses`, getting timeout errors.
Something like this:
`Test Front page URL: https://www.theguardian.com/europe TimeoutError: Waiting failed: 30000ms exceeded Stack: TimeoutError: Waiting failed: 30000ms exceeded`

Reply:
running `pbjs.getBidResponses()['dfp-ad--top-above-nav']` on the homepage for me returns undefined, are you sure it works?